python/ray/tune/README.rst

```diff
@@ -1,177 +1,6 @@
-        run: PPO
-        repeat: 2
-        stop:
-            episode_reward_mean: 200
-            time_total_s: 180
-        resources:
-            cpu: 5
-            driver_cpu_limit: 1  # of the 5 CPUs, only 1 is used by the driver
-        config:
-            num_workers: 4
-            timesteps_per_batch:
-                grid_search: [4000, 40000]
-            sgd_batchsize:
-                grid_search: [128, 256, 512]
-            num_sgd_iter:
-                eval: spec.config.sgd_batchsize * 2
-            lr:
-                eval: random.uniform(1e-4, 1e-3)
-
-When using the Python API, the above is equivalent to the following program:
-
-.. code:: python
-
-    import random
-    import ray
-    from ray.tune.result import TrainingResult
-    from ray.tune.trial_runner import TrialRunner
-    from ray.tune.variant_generator import grid_search, generate_trials
-
-    runner = TrialRunner()
-
-    spec = {
-        "env": "CartPole-v0",
-        "run": "PPO",
-        "repeat": 2,
-        "stop": {
-            "episode_reward_mean": 200,
-            "time_total_s": 180,
-        },
-        "resources": {
-            "cpu": 4,
-        },
-        "config": {
-            "num_workers": 4,
-            "timesteps_per_batch": grid_search([4000, 40000]),
-            "sgd_batchsize": grid_search([128, 256, 512]),
-            "num_sgd_iter": lambda spec: spec.config.sgd_batchsize * 2,
-            "lr": lambda spec: random.uniform(1e-4, 1e-3),
-        },
-    }
-
-    for trial in generate_trials(spec):
-        runner.add_trial(trial)
-
-    ray.init()
-
-    while not runner.is_finished():
-        runner.step()
-        print(runner.debug_string())
-
-Note that conditional dependencies between variables can be expressed by
-variable references, e.g. ``spec.config.sgd_batchsize`` in the above example.
-It is also possible to combine grid search and lambda functions by having
-a lambda function return a grid search object or vice versa.
-
-Using ray.tune as a library
----------------------------
-
-Ray.tune's Python API allows for finer-grained control over trial setup and
-scheduling. Some more examples of calling ray.tune programmatically include:
-
-- ``python/ray/tune/examples/tune_mnist_ray.py`` (see the main function)
-- ``python/ray/rllib/train.py``
-- ``python/ray/rllib/tune.py``
-
-Using ray.tune with Ray RLlib
------------------------------
-
-Another way to use ray.tune is through RLlib's ``python/ray/rllib/train.py``
-script. This script allows you to select between different RL algorithms with
-the ``--run`` option. For example, to train pong with the A3C algorithm, run:
-
-- ``./train.py --env=PongDeterministic-v4 --run=A3C --stop '{"time_total_s": 3200}' --resources '{"cpu": 8}' --config '{"num_workers": 8}'``
-
-or
-
-- ``./train.py -f tuned_examples/pong-a3c.yaml``
-
-You can find more RLlib examples in ``python/ray/rllib/tuned_examples``.
+https://github.com/ray-project/ray/blob/master/doc/source/tune.rst
```

Comment:
maybe: Documentation can be [found here](https://github.com/ray-project/ray/blob/master/doc/source/tune.rst)

Reply:
the url is not particularly pleasing ;)

Reply:
Done